### ISSUE TITLE
RouteOps.withAccessLog ignores implicit rejection handler

### ISSUE THREAD

**Issue by maciejbak85** [hseeberger/accessus#10]
Hi !

```
implicit def rejectionHandler =
    RejectionHandler.newBuilder().handleAll[MethodRejection] { rejections =>
      val methods = rejections map (_.supported)
      lazy val names = methods map (_.name) mkString ", "

      respondWithHeader(Allow(methods)) {
        options {
          complete(s"Supported methods : $names.")
        } ~
          complete(MethodNotAllowed, s"HTTP method not allowed, supported methods: $names!")
      }
    }
      .result()
```

and while using` .withAccessLog(accessLog(Logging(actorSystem, "ACCESS_LOG")))` it ignores rejection handler. Is there a way to use RH with accesslog ? Thanks!

**Comment by hseeberger** [comment]
`withAccessLog` is to be applied to the overall handler/route and `RejectionHandler`s are part of that. So any rejection should finally result in a `HttpResponse` and hence show up in the access log.

**Comment by maciejbak85** [comment]
well then it does not work this way.. :)

**Comment by hseeberger** [comment]
Could you please come up with a minified reproducer?

**Comment by maciejbak85** [comment]
```
 implicit def rejectionHandler =
    RejectionHandler.newBuilder().handleAll[MethodRejection] { rejections =>
      val methods = rejections map (_.supported)
      lazy val names = methods map (_.name) mkString ", "

      respondWithHeader(Allow(methods)) {
        options {
          complete(s"Supported methods : $names.")
        } ~
          complete(MethodNotAllowed, s"HTTP method not allowed, supported methods: $names!")
      }
    }
      .result()

```

```
def accessLog(log: LoggingAdapter): AccessLog[Long, Future[Done]] =
    Sink.foreach {
      case ((req, t0), res) =>
        val m = req.method.value
        val p = req.uri.path.toString
        val e = req.uri.query().toString()
        val s = res.status.intValue()
        val t = (now() - t0) / 1000000
        val ip = req.headers
          .collectFirst { case x: `X-Forwarded-For` => x }
          .map(_.addresses)
          .getOrElse("NA")
        log.info(s"$m $ip $p $e $s $t")
        log.info(req.headers.mkString(";"))
    }

 private def now() = System.nanoTime()
```

`Http().bindAndHandle(routes.rootRoutes.withAccessLog(accessLog(Logging(actorSystem, "ACCESS_LOG"))), host, port)`

**Comment by hseeberger** [comment]
Ah, now I understand. Thanks. You are right.

From the Akka docs:
>Once you have defined your custom RejectionHandler you have two options for “activating” it:
>
> 1. Bring it into implicit scope at the top-level
> 2. Supply it as an argument to the handleRejections directive

Please use 2. for the time being, because the implicit magic doesn't respect the implicitly defined rejection handler.

**Comment by hseeberger** [comment]
Should be fixed in v2.0.0 which should land at Maven Central eventually.
